### PR TITLE
fix(Combobox): add event propagation to mouse down

### LINKED ISSUE #9073: ComboBox not dismissing other open ComboBox or DropDown
## What package(s) are you using?

- [] `carbon-components`
- [x] `carbon-components-react`

## Detailed description
We have a page which has some ComboBox and DropDown objects. When a user opens the DropDown or a ComboBox then clicks on the chevron to open another ComboBox the open item is not dismissed

> Is this issue related to a specific component?
This is happening only with the ComboBox. If we open a ComboBox and click the DropDown the ComboBox dismisses.

> What did you expect to happen? What happened instead? What would you like to
> see changed?
Opening a ComboBox should dismiss any other open objects

> What browser are you working in?
Occurs in both Chrome and Firefox

> What version of the Carbon Design System are you using?
10.37.0 (was working as expected in 10.23.2)

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?
Product offering is IBM Planning Analytics

After
Before

- Screenshots or code
- Notes

### PR DESCRIPTION
**Pull request by dakahn**
Closes #9073 

- removes `event.stopPropagation()` from the `onMouseUp` event on the toggle button's props.
- `onMouseUp` on the toggle button now moves active focus to the text input

#### Testing / Reviewing

- go into `../ComboBox-story.js` and duplicate the `<ComboBox>` seen on line 65 like so
image
- open the **second** ComboBox's listbox either by clicking the text input or the toggle button
- click the **first** ComboBox's toggle button and observe the **second** Combobox's listbox close as expected

#### Notes
@joshblack pretty sure stopping event propagation was critical for some reason we're missing here -- do you remember why?

### CODE REVIEW

**Review by tw15egan — APPROVED** [review]
Working as expected 👍🏻 ✅

File: packages/react/src/components/ComboBox/ComboBox.js [hunk 1]
```diff
@@ -255,8 +255,10 @@ const ComboBox = React.forwardRef((props, ref) => {
           // stop the event from propagating to prevent this. This allows the
           // toggleMenu behavior for the toggleButton to correctly open and
           // close the menu.
-          onMouseUp(event) {
-            event.stopPropagation();
+          onMouseUp() {
```
Comment (joshblack) [inline packages/react/src/components/ComboBox/ComboBox.js]: Does the above comment still apply? Seems like we still might be running into this without the propagation (first demo is deploy preview, second is from stable)

https://user-images.githubusercontent.com/3901764/128525519-bfe0438d-3544-4d17-9d82-bb054c8cfe0f.mov

**Review by joshblack — CHANGES_REQUESTED** [review]

File: packages/react/src/components/ComboBox/ComboBox.js (hunk 1 above)
Comment (dakahn) [reply packages/react/src/components/ComboBox/ComboBox.js]: ok cool, i didn't really understand the comment 100%, but now its clear whats going on there. So what's the workaround?

File: packages/react/src/components/ComboBox/ComboBox-test.js [hunk 2]
```diff
@@ -201,5 +203,45 @@ describe('ComboBox', () => {
       const wrapper = mount(<ComboBox {...mockProps} />);
       expect(wrapper.find('input').instance().value).toBe('');
     });
+
+    it('should only render one listbox at a time when multiple comboboxes are present', () => {
```
Comment (tay1orjones) [inline packages/react/src/components/ComboBox/ComboBox-test.js:207]: I also added a new test here to cover this functionality with multiple instances of combobox on the same page. 

Maybe we don't need the extra `multiple instances` story with this test in place.

**Review by joshblack — APPROVED** [review]

**Review by tw15egan — APPROVED** [review]
Re-reviewed and removed the test story, LGTM 👍🏻

**Pull request merged**